code_examples/sdk_examples/src/core_features/utils/generateKeypairs.ts

```diff
@@ -1,55 +1,33 @@
 import * as Kilt from '@kiltprotocol/sdk-js'
 
-import {
-  blake2AsU8a,
-  keyExtractPath,
-  keyFromPath,
-  mnemonicGenerate,
-  mnemonicToMiniSecret,
-  sr25519PairFromSeed
-} from '@polkadot/util-crypto'
+import { mnemonicGenerate } from '@polkadot/util-crypto'
 
-// Because there is no first-class support for this class of keys,
-// we need to use a workaround to generate a key we can use for encryption/decryption.
-function generateKeyAgreement(mnemonic: string): Kilt.KiltEncryptionKeypair {
-  const secretKeyPair = sr25519PairFromSeed(mnemonicToMiniSecret(mnemonic))
-  const { path } = keyExtractPath('//did//keyAgreement//0')
-  const { secretKey } = keyFromPath(secretKeyPair, path, 'sr25519')
-  return Kilt.Utils.Crypto.makeEncryptionKeypairFromSeed(blake2AsU8a(secretKey))
-}
+const signingKeyPairType = 'sr25519'
 
 export function generateKeypairs(mnemonic = mnemonicGenerate()): {
-  authentication: Kilt.KiltKeyringPair & {
-    type: 'sr25519'
-  }
+  authentication: Kilt.KiltKeyringPair
   keyAgreement: Kilt.KiltEncryptionKeypair
   assertionMethod: Kilt.KiltKeyringPair
   capabilityDelegation: Kilt.KiltKeyringPair
 } {
-  const keyring = new Kilt.Utils.Keyring({
-    ss58Format: 38,
-    type: 'sr25519'
-  })
-  const account = keyring.addFromMnemonic(mnemonic)
-
-  const authentication = {
-    ...account.derive('//did//0'),
-    type: 'sr25519'
-  } as Kilt.KiltKeyringPair & {
-    type: 'sr25519'
-  }
-
-  const assertionMethod = {
-    ...account.derive('//did//assertion//0'),
-    type: 'sr25519'
-  } as Kilt.KiltKeyringPair
-
-  const capabilityDelegation = {
-    ...account.derive('//did//delegation//0'),
-    type: 'sr25519'
-  } as Kilt.KiltKeyringPair
-
-  const keyAgreement = generateKeyAgreement(mnemonic)
+  const authentication = Kilt.Utils.Crypto.makeKeypairFromUri(
```

Comment:
I can remove the constant. Any reason why you would put it as a string over constant?

Reply:
You would need to look up where the variable comes from. Having the string there makes it easier to see immediately what we put in there.

I looked at the code on my phone and needed to search where the variable `signingKeyPairType` comes from.

Reply:
> This would distinguish the three different keys. We could have three different mnemonics to provide more clarity if we wanted to.

I would use different mnemonics or keep the derivation part. But doing the same thing three times looks wrong.